user-domain/src/main/java/nl/vpro/domain/user/UserService.java

```diff
@@ -29,8 +29,17 @@
  */
 public interface UserService<T extends User> {
 
+    ThreadPoolExecutor ASYNC_EXECUTOR =
```

Comment:
Waarom niet de andere async call gebruiken die ik had toegevoegd en eigen threadpool voor upload. Gezien de duur waarmee uploads kunnen hangen lijkt threads delen me niet ideaal

Reply:
Het lijkt me juist wel goed om threads te delen. Maar in elk geval is het al fout dus om voor de call zonder explciete execute default die backgroundexecutor te gebruiken.

als je deze gebruikt heb je een threadpool voor alles was je 'async' wil uitvoeren als een zekere gebruiker. Als de pool te klein is zal hij er meer threads in zetten. Ik zie eigenlijk geen nadeel.

Reply:
als er threads blijven hangen, al de threadpool alsmaar blijven groeien dus. 

Dat is fout, maar in elk geval duurt het wel even voordat er echt iets niet meer werkt dan. Als je een eigen thread-pool gebruikt, zal het voor die alleen gelden, maar qua resources lijkt me dat dat niks uit maakt. Er zit nu een maximum van 10000 threads op. Als die allemaal vol zitten met dingen die blijven 'hangen' op dit probleem , dan lijkt het me dat we wel een groter probleem hebben dan dat er niks meer asynchroon uitgevoerd wordt.

Reply:
Maar nu kun je niet de blast radius van een trage upload server beperken.

Daarnaast heet LinkedBlockingDeque max queue size Integer.MAX_VALUE, dus ook hier zal altijd netto 1 thread beschikbaar zijn toch? Alle volgende bezoeken wordt gequeud tot int max

Reply:
Ok, wat mij betreft is die nieuwe async call ook niet zinloos, en we kunnen hem hier gebruiken, maar niet niet weg dat die normale async-call ook op verschillende plekken gebruikt wordt denk ik. Dus het moet gewoon goed werken. Ik geloof inderdaad dat het nog niet werkte hoe ik dacht, maar nu wel.